File: src/me/krickl/memebotj/InternalCommands/AdminCommands/RestartThreadCommand.scala
package me.krickl.memebotj.InternalCommands.AdminCommands

import me.krickl.memebotj.Memebot

/**
 * Created by unlink on 21/11/15.
 */
class RestartThreadCommand(channel: String, command: String = "null", dbprefix: String = "") extends me.krickl.memebotj.CommandHandler(channel, command, dbprefix) {
  this.setNeededCommandPower(75)

  override def commandScript(sender: me.krickl.memebotj.UserHandler, channelHandler: me.krickl.memebotj.ChannelHandler, data: Array[String]) = {
    channelHandler.sendMessage(Memebot.formatText(channelHandler.localisation.localisedStringFor("RESTART"), channelHandler, sender, this, false, Array()), this.getChannelOrigin)
    channelHandler.writeDBChannelData()
    channelHandler.setIsJoined(false)
  }
}

File: src/main/java/me/krickl/memebotj/Commands/Internal/RestartThreadCommand.java
package me.krickl.memebotj.Commands.Internal;

import me.krickl.memebotj.ChannelHandler;
import me.krickl.memebotj.Commands.CommandHandler;
import me.krickl.memebotj.Memebot;
import me.krickl.memebotj.UserHandler;
import me.krickl.memebotj.Utility.CommandPower;

/**
 * This file is part of memebotj.
 * Created by unlink on 11/04/16.
 */
public class RestartThreadCommand extends CommandHandler {
    public RestartThreadCommand(ChannelHandler channelHandler, String commandName, String dbprefix) {
        super(channelHandler, commandName, dbprefix);
    }

    @Override
    public void overrideDB() {
        this.setNeededCommandPower(CommandPower.adminAbsolute);
    }

    @Override
    public void commandScript(UserHandler sender, String[] data) {
        getChannelHandler().sendMessage(Memebot.formatText(getChannelHandler().getLocalisation().localisedStringFor("RESTART"), getChannelHandler(), sender, this, false, new String[]{}, ""), this.getChannelHandler().getChannel());
        Memebot.saveAll();
        getChannelHandler().setJoined(false);
    }
}
